src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -642,6 +659,68 @@ bool AreaTrigger::HasSplines() const
     return bool(_spline);
 }
 
+void AreaTrigger::InitCircularMovement(AreaTriggerCircularMovementInfo const& cmi, uint32 timeToTarget)
+{
+    // should be sent in object create packets only
+    m_uint32Values[AREATRIGGER_TIME_TO_TARGET] = timeToTarget;
+
+    _areaTriggerCircularMovementInfo = cmi;
+
+    _areaTriggerCircularMovementInfo.TimeToTarget = GetTimeToTarget();
+    _areaTriggerCircularMovementInfo.ElapsedTimeForMovement = GetElapsedTimeForMovement();
+
+    if (IsInWorld())
+    {
+        // TODO: broadcast AreaTriggerReShape?
+    }
+}
+
+Position const* AreaTrigger::GetCircularMovementCenterPosition() const
+{
+    if (_areaTriggerCircularMovementInfo.TargetGUID.is_initialized())
+        if (WorldObject* center = ObjectAccessor::GetWorldObject(*this, *_areaTriggerCircularMovementInfo.TargetGUID))
+            return center;
+
+    return _areaTriggerCircularMovementInfo.Center.is_initialized() ? &_areaTriggerCircularMovementInfo.Center->Pos : nullptr;
+}
+
+void AreaTrigger::UpdateCircularMovementPosition()
+{
+    Position const* centerPos = GetCircularMovementCenterPosition();
+    if (!centerPos)
+        return;
+
+    AreaTriggerCircularMovementInfo& cmi = _areaTriggerCircularMovementInfo;
+    if (cmi.StartDelay > GetElapsedTimeForMovement())
```

Comment:
i guess the at will skip the first part of the circle path

Reply:
What you mean? If i set startdelay to 5000, the areatrigger will wait 5 seconds clientside, before he starts to move

Reply:
if one circle takes 20 sec, the at starts at angle of pi/2

Reply:
you're correct.. will change that